PurchasesTests/SubscriberAttributes/BackendSubscriberAttributesTests.swift

```diff
@@ -0,0 +1,231 @@
+        let underlyingError = NSError(domain: "domain", code: 0, userInfo: nil)
+
+        mockHTTPClient.stubbedCompletionError = Purchases.ErrorUtils.networkError(withUnderlyingError: underlyingError)
+
+        var receivedError: Error? = nil
+        backend.postSubscriberAttributes([
+                                             subscriberAttribute1.key: subscriberAttribute1,
+                                             subscriberAttribute2.key: subscriberAttribute2
+                                         ],
+                                         appUserID: appUserID,
+                                         completion: { (error: Error!) in
+                                             completionCallCount += 1
+                                             receivedError = error
+                                         })
+
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 1
+        expect(completionCallCount).toEventually(equal(1))
+        expect(receivedError).toNot(beNil())
+        expect(receivedError).to(beAKindOf(Error.self))
+        expect((receivedError! as NSError).code) == Purchases.ErrorCode.networkError.rawValue
+    }
+
+    func testPostSubscriberAttributesCallsCompletionWithErrorInBackendErrorCase() {
+        var completionCallCount = 0
+        mockHTTPClient.shouldInvokeCompletion = true
+        mockHTTPClient.stubbedCompletionStatusCode = 503
+        mockHTTPClient.stubbedCompletionError = nil
+
+        var receivedError: Error? = nil
+        backend.postSubscriberAttributes([
+                                             subscriberAttribute1.key: subscriberAttribute1,
+                                             subscriberAttribute2.key: subscriberAttribute2
+                                         ],
+                                         appUserID: appUserID,
+                                         completion: { (error: Error!) in
+                                             completionCallCount += 1
+                                             receivedError = error
+                                         })
+
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 1
+        expect(completionCallCount).toEventually(equal(1))
+        expect(receivedError).toNot(beNil())
+        expect(receivedError).to(beAKindOf(Error.self))
+
+        let receivedNSError = receivedError! as NSError
+        expect(receivedNSError.code) == Purchases.ErrorCode.unknownBackendError.rawValue
+    }
+
+    func testPostSubscriberAttributesNoOpIfAttributesAreEmpty() {
+        var completionCallCount = 0
+        backend.postSubscriberAttributes([:],
+                                         appUserID: appUserID,
+                                         completion: { (error: Error!) in
+                                             completionCallCount += 1
+                                         })
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 0
+    }
+
```

Comment:
I would add some tests regarding the `RCShouldMarkSyncedKey` stuff. I think something like this should be enough:
1. On `503` it is false (you can add the assert to the existing test)
2. On  `304` it is true

Reply:
good catch, added